wp_serde_helper/src/lib.rs

```diff
@@ -129,6 +174,45 @@ where
     }
 }
 
+struct DeserializeFalseOrStringVisitor;
+
+impl de::Visitor<'_> for DeserializeFalseOrStringVisitor {
+    type Value = Option<String>;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("Boolean `false` or a string")
+    }
+
+    fn visit_bool<E>(self, v: bool) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        if v {
+            Err(E::invalid_value(Unexpected::Bool(v), &self))
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        if v.to_lowercase().trim() == "false" {
+            return Ok(None);
+        }
+
+        Ok(Some(v.to_string()))
+    }
+}
```

Comment:
I think implementing a Visitor is a bit overkill here. An alternative could be something like this:

```rust
enum Value {
  String { value: String }
  Bool { value: bool }
}

fn deserialize_false_or_string() {
  let value = Value::deserialize(deserializer)
  match value {
    ...
  }
}
```

---

BTW, I wonder if the `WpDeserialize` macro might be helpful in parsing WpApiDetails.

Reply:
`WpDeserialize` would only work if we mark all fields as `Option`. Otherwise, we have to manually implement `WpDeserialize` does.

`WpDeserialize` uses `DeserializeEmptyVecOrT` which doesn't handle `false` or `String`, so it wouldn't be helpful even if we could use `WpDeserialize` for `WpApiDetails`.

Reply:
> I think implementing a Visitor is a bit overkill here. An alternative could be something like this

I think it's fine to implement a Visitor for this. It's more performant and the implementation is not particularly complicated.